src/components/ArticleView.jsx

```diff
@@ -0,0 +1,53 @@
+import { useParams } from "react-router-dom";
+import { useState, useEffect } from "react";
+import { getArticleById } from "../utils/api";
+
+const ArticleView = () => {
+  const { article_id } = useParams();
+  const [article, setArticle] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [err, setErr] = useState("");
+
+  useEffect(() => {
+    setIsLoading(true);
+    getArticleById(article_id)
+      .then((articleFromApi) => {
+        if (articleFromApi) {
+          setArticle(articleFromApi);
+          setIsLoading(false);
+        } else {
+          setErr("Article not found.");
+          setIsLoading(false);
+        }
```

Comment:
There is no need for the conditional logic. You set your Api up so that this endpoint will respond with an article if it exists. If it doesn't, it will respond with an error which will automatically fall into your catch block 😄

Reply:
You're so right, thank you Rayhaan!